Remove deprecated SingleStateFeaturizer subclasses and method. (#9171)
RasaHQ/rasa: changelog/8879.removal.md

> Follow through on removing deprecated dialogue policy state featurizers: `BinarySingleStateFeature` and `LabelTokenizerSingleStateFeaturizer`.

Reviewer: In the interest of simple and kind of consistent language across our changelog:
```suggestion
Removed the deprecated dialogue policy state featurizers: `BinarySingleStateFeature` and `LabelTokenizerSingleStateFeaturizer`.
```

> Remove deprecated `SingleStateFeaturizer` method `encode_all_actions`. Use `encode_all_labels` instead.

Reviewer: ```suggestion
Removed the deprecated method `encode_all_actions` of `SingleStateFeaturizer`. Use `encode_all_labels` instead.
```
